src/module/__tests__/helpers-dice.test.ts

```diff
@@ -0,0 +1,484 @@
+/**
+ * @file Contains tests for dice helpers
+ */
+// eslint-disable-next-line prettier/prettier, import/no-cycle
+import { QuenchMethods } from "../../e2e";
+import { closeDialogs, trashChat, waitForInput } from "../../e2e/testUtils";
+import OseDice from "../helpers-dice";
+
+export const key = "ose.helpers.dice";
+export const options = {
+  displayName: "Helpers: Dice",
+};
+
+const createMockData = (type: string, target: number, thac0: number = 0) => ({
+  roll: { type, target, thac0 },
+});
+
+const createMockRoll = (target: number, results: Array<number> = [target]) => ({
+  terms: [{ total: target, results }],
+  total: target,
+});
+
+const createMockAttackData = () => ({
+  parts: ["20"],
+  data: {
+    roll: {
+      blindroll: false,
+      dmg: ["1d6"],
+      thac0: 15,
+      target: {
+        actor: { system: { ac: { value: 0 }, aac: { value: 9 } } },
+      },
+    },
+  },
+});
+
+export default ({
+  describe,
+  it,
+  after,
+  afterEach,
+  before,
+  expect,
+}: QuenchMethods) => {
+  const acSetting = game.settings.get(game.system.id, "ascendingAC");
+
+  before(async () => {
+    await ui.notifications?.close();
+  });
+  after(async () => {
+    game.settings.set(game.system.id, "ascendingAC", acSetting);
+    await ui.notifications?.render(true);
+  });
+
+  describe("sendRoll(parts, data, title, flavor, speaker, form, chatMessage)", () => {
+    before(async () => {
+      await trashChat();
+    });
+    it("Can roll with single part", async () => {
+      const dice = await OseDice.sendRoll({
+        parts: ["1d10"],
+        data: { roll: { blindroll: false } },
+      });
+      await waitForInput();
+      await waitForInput();
+      const chatCard = document.querySelector(".chat-message .dice-formula")?.innerHTML;
+      expect(Object.keys(dice)).contain("_evaluated");
+      expect(dice._evaluated).equal(true);
+      expect(Object.keys(dice)).contain("_formula");
+      expect(dice._formula).equal("1d10");
+      expect(chatCard).equal("1d10");
+    });
+    it("Can roll with multiple parts", async () => {
+      const dice = await OseDice.sendRoll({
+        parts: ["1d10", "1d20"],
+        data: { roll: { blindroll: false } },
+      });
+      await waitForInput();
+      await waitForInput();
+      const chatCard = document.querySelector(".chat-message .dice-formula")?.innerHTML;
+      expect(Object.keys(dice)).contain("_evaluated");
+      expect(dice._evaluated).equal(true);
+      expect(Object.keys(dice)).contain("_formula");
+      expect(dice._formula).equal("1d10 + 1d20");
+      expect(chatCard).equal("1d10 + 1d20");
+    });
+    if (game.dice3d) {
+      describe("Dice So Nice integration", () => {});
```

Comment:
Another one found, not sure if a to-do?

Reply:
No, it was for discussion but I forgot to mark them!